packages/web/src/hooks/common/use-component-size.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect, useRef } from "react";
+
+interface ComponentSize {
+  width: number;
+  height: number;
+}
+
+function useComponentSize(): [React.RefObject<HTMLDivElement>, ComponentSize] {
+  const [size, setSize] = useState<ComponentSize>({ width: 0, height: 0 });
+  const componentRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleResize = () => {
+      const { width, height } =
+        componentRef.current?.getBoundingClientRect() ?? {
+          width: 0,
+          height: 0,
+        };
+      setSize({ width, height });
+    };
+
+    handleResize();
+
+    window.addEventListener("resize", handleResize);
+
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
```

Comment:
It would be better if we could detect changes in the state of the `componentRef`.

Reply:
Edited because I thought it was a good comment.